### ISSUE TITLE
[Horizontal rule]: QA: Horizontal rule is not visually available on high contrast themes

### ISSUE THREAD

**Issue by gopinathan-n** [carbon-design-system/carbon-for-ibm-dotcom#9712]
### Description

Page Element:
Separator present after the heading

Issue Description:
Horizontal rule is not visually available on high contrast black and white.

Recommend Fix:
Ensure that the horizontal rule is visually available on applying high contrast.

User Impact:
After high contrast is activated, focus indicators and component interactions & defined component colors remain visible via keyboard

### Component(s) impacted

Horizontal rule
Canary
Staging

### Browser

Firefox

### Carbon for IBM.com version

v1.40.0-rc.0

### Severity

Severity 1 = The design is broken in a critical way that blocks users from completing tasks or damages the brand. Affects major functionality, no workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components-react

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
